cli/azd/pkg/rzip/rzip.go

```diff
@@ -6,55 +6,129 @@ package rzip
 import (
 	"archive/zip"
 	"io"
-	"io/fs"
+	"log"
 	"os"
 	"path/filepath"
 	"strings"
 )
 
+// CreateFromDirectory creates a zip archive from the given directory recursively,
+// that is suitable for transporting across machines.
+//
+// It resolves any symlinks it encounters.
 func CreateFromDirectory(source string, buf *os.File) error {
 	w := zip.NewWriter(buf)
-	err := filepath.WalkDir(source, func(path string, info fs.DirEntry, err error) error {
-		if err != nil {
-			return err
-		}
 
-		if info.IsDir() {
-			return nil
-		}
-		fileInfo, err := info.Info()
-		if err != nil {
-			return err
-		}
+	err := addDirRoot(w, source)
+	if err != nil {
+		return err
+	}
 
-		header := &zip.FileHeader{
-			Name: strings.Replace(
-				strings.TrimPrefix(
-					strings.TrimPrefix(path, source),
-					string(filepath.Separator)), "\\", "/", -1),
-			Modified: fileInfo.ModTime(),
-			Method:   zip.Deflate,
-		}
+	return w.Close()
+}
 
-		f, err := w.CreateHeader(header)
+func addDirRoot(
+	w *zip.Writer,
+	src string) error {
+	return addDir(w, "", src, 0)
+}
+
+func addDir(
+	w *zip.Writer,
+	destRoot,
+	src string,
+	symlinkDepth int) error {
+	if symlinkDepth > 40 {
+		// too deep, bail out similarly to the 'zip' tool
+		log.Println("skipping", src, "too many levels of symbolic links")
+		return nil
+	}
```

Comment:
Just curious - Is `40` some well-known number to bail out?

Reply:
yeah, it's `MAXSYMLINKS` in the linux kernel